androidApp/build.gradle.kts

```diff
@@ -9,7 +9,7 @@ android {
     compileSdk = 33
     defaultConfig {
         applicationId = "co.nimblehq.avishek.phong.kmmic.android"
-        minSdk = 21
+        minSdk = 23
```

Comment:
I actually asked [this](https://nimble-co.slack.com/archives/C051TNKTDFB/p1682052334167079?thread_ts=1682047801.791059&cid=C051TNKTDFB) after seeing this change here. Since JSON API now supports 21, can't we keep it 21 here as well? 🤔

Reply:
@AVI5HEK, I don't have any problem with this. However, the `shared` module minSdk is 23, and the Android app which consumes the `shared` module has minSdk=21; it can be a compatibility issue that is unsurfaced. What do you think if I keep the minSdk = 23 here?

Reply:
@phongvhd93 Are you talking about this one? It's set to `21` in the `:shared` module. 🤔 
![Uploading image.png…]()

Reply:
@AVI5HEK, oh sorry, I forgot that we updated the JSON-API to use the one from Nimble's not from the fork from my repo

Reply:
MinSDK must be 23 to use the JSON API. I also updated the `build.gradle.kts` in this PR

![Screenshot 2023-04-24 at 17 15 32](https://user-images.githubusercontent.com/22606906/233968446-5bc75576-cad0-460d-850d-be0321bc0328.png)

Reply:
Oh, I see. Let's go with 23 then. ☹️